### PR TITLE
Convert files on e2e folder from Java to Kotlin

### PR DESCRIPTION
**Pull request by jostnes**
### What does this PR do
This PR converts files in `e2e` folder from Java to Kotlin. Initially, I attempted to convert all the files in `androidTest` but that exercise ended up with many errors that I wasn't sure how to fix as I'm unfamiliar with the connected tests that are not UI tests. This PR focuses only on the `e2e` folder. With this change, the UI tests would now have access to Compose test API without using a workaround. 

This PR has multiple commits as they represent the file conversion (for some files, if there are no errors, I combine a few in the same commit). Commits that do not start with "convert" are commits to fix errors that appeared after the conversion. This is to help reviewers to be able to distinguish between changes that were automagically done by Android Studio and changes that I made to remove those errors. 

### Testing
All instrumented tests should work as expected locally and in CI.

### CODE REVIEW

**Comment by pachlava** [comment]
While I did not do a review yet, I'm confused (positively) by two things. They most probably are not related to the changes, but are caused by some changes in Buildkite 🤔 

1. The Instrument Tests task now takes 7–8 minutes less, because the app build time (before it gets sent to FTL) went from ~12 minutes to ~5. The tests execution time is not affected though.
2. The `e2eAllDayStatsLoad` does not fail here, as well as it stopped failing in #18074 which is based on this branch.

**Comment by pachlava** [comment]
> The `e2eAllDayStatsLoad` does not fail here, as well as it stopped failing in https://github.com/wordpress-mobile/WordPress-Android/pull/18074 which is based on this branch.

Taking my words back, it [failed](https://buildkite.com/automattic/wordpress-android/builds/9815#0186c629-219d-4ba7-9b68-b8b9410b7949) for JP.

File: WordPress/src/androidTest/java/org/wordpress/android/e2e/pages/PostsListPage.kt
Comment (pachlava) [inline WordPress/src/androidTest/java/org/wordpress/android/e2e/pages/PostsListPage.kt]: I was a bit confused by this removal, and then saw that it was caused by having a `Condition 'recyclerView == null' is always 'false'` warning otherwise. It looks like the value of `recyclerView` could be `null` in times of Java, but it'll never be `null` in Kotlin? 🤔 

Regardless of this, `PostsListPage` is only used in `WPScreenshotTest` in `editBlogPost` function, which is not used, so I guess we're fine with this adjustment 🤷

**Review by pachlava — DISMISSED** [review]
Thanks for this impressive amount of work in improving the framework consistency, @jostnes!

I went through the commits, and the only remark I had is this one. Given the fact that code mentioned in remark it's not in use, and that CI is green, it's good to go from my side :shipit: 👍

File: WordPress/src/androidTest/java/org/wordpress/android/e2e/pages/PostsListPage.kt
Comment (pachlava) [inline WordPress/src/androidTest/java/org/wordpress/android/e2e/pages/PostsListPage.kt]: I'm sorry for approving and then dismissing the review @jostnes, there is something I noticed about this part already after I approved the PR.

In `wPScreenshotTest` I noticed that `editBlogPost` is disabled (making the removed part of code not really affect anything), but then I noticed a [comment](https://github.com/wordpress-mobile/WordPress-Android/blob/update-e2e-files-to-kotlin/WordPress/src/androidTest/java/org/wordpress/android/ui/screenshots/WPScreenshotTest.java#L78) above the disabled `editBlogPost` call saying that it's disabled because it fails on CI.

This made me think that it's still executed locally (this is where screenshot tests are usually executed, IINM). After trying to execute it, it appeared that the `recyclerView` locator that was left after removal from above did not work, and additionally it failed the run, because of "can't cast `null` to RecyclerView". Returning back the removed code and applying a minor change (moving `as RecyclerView` to `return`) did the trick - the secondary locator now is called:

```kotlin
    // Workaround for cases when recyclerview id missing
    private val recyclerView: RecyclerView?
        get() {
            val pager = WPSupportUtils.getCurrentActivity().findViewById<ViewPager>(R.id.postPager)
            var recyclerView = pager.getChildAt(pager.currentItem)
                .findViewById<View>(R.id.recycler_view)
            if (recyclerView == null) {
                // Workaround for cases when recyclerview id missing
                recyclerView = ((pager.getChildAt(pager.currentItem)
                    .findViewById<View>(R.id.ptr_layout) as ViewGroup).getChildAt(0) as ViewGroup).getChildAt(
                    0
                )
            }
            return recyclerView as RecyclerView
        }
```

**Review by pachlava — CHANGES_REQUESTED** [review]
Hi again @jostnes 😅 👋 

I left an extended [comment](https://github.com/wordpress-mobile/WordPress-Android/pull/18071#discussion_r1131006709) where I describe the reason for asking for a change. There might be some overthinking involved, but this way we stay on the safer side 🙇 In other aspects the PR is good to go from my side.

File: WordPress/src/androidTest/java/org/wordpress/android/e2e/pages/PostsListPage.kt
Comment (ParaskP7) [reply WordPress/src/androidTest/java/org/wordpress/android/e2e/pages/PostsListPage.kt]: 👋 @pachlava @jostnes !

Actually, I think the issue was that the conversion of this file from Java to Kotlin, which left the `RecyclerView` cast as non-null, instead of it being nullable (I mean here `findViewById<View>(R.id.recycler_view)`).

As such, just adding a `?` on `RecyclerView` and making it `RecyclerView?` would all that needed happened here (I think). 

End result:

```kotlin
    // Workaround for cases when recyclerview id missing
    private val recyclerView: RecyclerView?
        get() {
            val pager = WPSupportUtils.getCurrentActivity().findViewById<ViewPager>(R.id.postPager)
            var recyclerView = pager.getChildAt(pager.currentItem)
                .findViewById<View>(R.id.recycler_view) as RecyclerView?
            if (recyclerView == null) {
                // Workaround for cases when recyclerview id missing
                recyclerView = ((pager.getChildAt(pager.currentItem)
                    .findViewById<View>(R.id.ptr_layout) as ViewGroup).getChildAt(0) as ViewGroup).getChildAt(
                    0
                ) as RecyclerView
            }
            return recyclerView
        }
```

Let me know if that makes sense to you. 🙏

**Comment by ParaskP7** [comment]
👋 @jostnes and @pachlava (as an FYI to you too), just in case it will help you in a future such work.

Notice the below change on one of your Kotlinization commits (just as an example):

image

This change makes it difficult to review as the old Java file is deleted and a new Kotlin file is created. One would need to manually compare them and this, unfortunately, takes extra time.

-----

What you can do is committing your change by checking this `Extra commit for .java > .kt renames` option:

Screenshot 2023-03-09 at 3 57 53 PM

Now notice the difference:

Screenshot 2023-03-09 at 4 00 35 PM

This change now makes it easier to review as the old Java and new Kotlin file can be diffed.

Does that help? 🙏 

Let me know and I can expand on that if you like! 💯

File: WordPress/src/androidTest/java/org/wordpress/android/e2e/pages/PostsListPage.kt
Comment (jostnes) [reply WordPress/src/androidTest/java/org/wordpress/android/e2e/pages/PostsListPage.kt]: @pachlava @ParaskP7 Thanks for pointing this out! I removed it as suggested by Android Studio as the condition was never going to be met. After the change, I saw that everything was still passing, so I incorrectly assumed that everything was good. 

I've reverted the if condition and made RecyclerView nullable by adding the `?` as suggested. Updated in https://github.com/wordpress-mobile/WordPress-Android/pull/18071/commits/c2c88c71aa63519c0fe5c2d3488aad62cb92068f

**Comment by jostnes** [comment]
Thanks @ParaskP7 for sharing that! I found it quite hard to review my own changes, too, as one file was deleted and a new one was added. I was also looking at [past Kotlinization efforts](https://github.com/wordpress-mobile/WordPress-Android/issues/14904) and the way is the same as I did in this PR so knowing this new way is going to be really helpful for both author and reviewers!

> What you can do is committing your change by checking this Extra commit for .java > .kt renames option:
I don't use git from within Android Studio so unfortunately I'm not very familiar with it, I'll explore this option and let you know separately if I have any issues getting it to work. But from what you shared, I definitely agree this is a better way for java to kotlin changes!

For this PR, do you think it's worth changing it to that (use the "rename" method instead of the "delete .java, add .kt" way)? Or okay to keep this PR as is and do that moving forward?

**Comment by jostnes** [comment]
Thanks @pachlava for the review! For `e2eAllDayStatsLoad` looks like it's flaky issue instead of a permanent failure, from the last commit, I retried once and it passed 🤔 

I saw that you're already working on it via https://github.com/wordpress-mobile/WordPress-Android/pull/18074, so I can safely leave that out of this PR. Thanks for working on that!

That said, this PR shouldn't contain functional changes only file conversion, so those changes you saw are most likely coincidence 😄

**Review by pachlava — APPROVED** [review]
Thank you for the c2c88c71aa63519c0fe5c2d3488aad62cb92068f @jostnes 👋 🙂 !

It's good to go from my side 👍

File: WordPress/src/androidTest/java/org/wordpress/android/e2e/pages/PostsListPage.kt
Comment (ParaskP7) [reply WordPress/src/androidTest/java/org/wordpress/android/e2e/pages/PostsListPage.kt]: Awesome, thanks for the update @jostnes ! 🙇 ❤️ 🚀

**Comment by ParaskP7** [comment]
👋 @jostnes !

> Thanks @ParaskP7 for sharing that! I found it quite hard to review my own changes, too, as one file was deleted and a new one was added. I was also looking at https://github.com/wordpress-mobile/WordPress-Android/issues/14904 and the way is the same as I did in this PR so knowing this new way is going to be really helpful for both author and reviewers!

💯 Yea, in the past one would had to do that manually in order to get the same result, and not many of us did go through that process. But, nowadays AS is doing that for us automagically, so it is easy to take advantage of it! 💯 

> For this PR, do you think it's worth changing it to that (use the "rename" method instead of the "delete .java, add .kt" way)? Or okay to keep this PR as is and do that moving forward?

I'll leaving it up to you, maybe you would like to redo this PR and see how this this `Extra commit for .java > .kt renames` option works and then do another code review on those changes yourself. It might be a good occasion for you to get some exposure on some more Jave-to-Kotlin conversions as I am envisioning you to be continuing towards this path when possible.

But, I am leaving that decision up to you and how comfortable you are in re-doing this work. It is a 👍 from my side no matter. 🥇

**Comment by jostnes** [comment]
Thanks @ParaskP7 for the recommendation! In that case, I'll leave this open over the weekend and decide on Monday after planning my week. This for now is only a subset of `AndroidTest` so there are other areas that _will_ need to go through this Kotlinization process too (and for those other areas will definitely be using the suggested way 😄)

**Comment by jostnes** [comment]
I'm going to merge this for now. Future Kotlinization process will follow the suggested way for easier reviews. cc: @ParaskP7

**Pull request merged**
